report/validation/thermodesorption_spectra/dark/dark_tds.md

```diff
@@ -0,0 +1,256 @@
+        E_k = E_D,
+        p_0 = 1e13,
+        E_p = E_p[i],
+        density = n_i[i] * damage_dist,
+        materials = damaged_tungsten,
+    ))
+
+traps = [trap_1] + neutron_induced_traps
+model.traps = traps
+
+model.dt = F.Stepsize(
+    initial_value=1,
+    stepsize_change_ratio=1.1,
+    t_stop=t_imp + t_rest * 0.5,
+    dt_min=1e-1,
+    stepsize_stop_max=50,
+)
+
+model.settings = F.Settings(
+    absolute_tolerance=1e10,
+    relative_tolerance=1e-10,
+    final_time=start_tds + (max_temp - min_temp) / Beta,  # time to reach max temp
+)
+
+derived_quantities = F.DerivedQuantities(
+    [
+        F.AverageVolume("T", volume=1),
+        F.TotalVolume("solute", volume=1),
+        F.TotalVolume("retention", volume=1),
+        F.TotalVolume("1", volume=1),
+        F.TotalVolume("2", volume=1),
+        F.TotalVolume("3", volume=1),
+        F.TotalVolume("4", volume=1),
+        F.TotalVolume("5", volume=1),
+        F.TotalVolume("6", volume=1),
+        F.HydrogenFlux(surface=1),
+        F.HydrogenFlux(surface=2),
+    ],
+)
+
+model.exports = [derived_quantities]
+
+model.initialise()
+model.run()
+```
+
+```{note}
+Implementation details from {cite}`dark_modelling_2024_code`.
+```
+
++++
+
+## Comparison with experimental data
+
+The results produced by FESTIM are in good agreement with the experimental data. The grey areas represent the contribution of each trap to the global TDS spectrum.
+
+```{code-cell} ipython3
+:tags: [hide-input]
+
+t = derived_quantities.t
+flux_left = derived_quantities.filter(fields="solute", surfaces=1).data
+flux_right = derived_quantities.filter(fields="solute", surfaces=2).data
+flux_total = -np.array(flux_left) - np.array(flux_right)
+
+t = np.array(t)
+temp = min_temp + Beta * (t - start_tds)
+
+# plotting simulation data
+plt.plot(temp, flux_total, linewidth=3, label="FESTIM")
+
+# plotting trap contributions
+trap_data = [derived_quantities.filter(fields=f"{i}").data for i in range(1, 7)]
+contributions = [-np.diff(trap) / np.diff(t) for trap in trap_data]
+
+colors = [(0.9*(i % 2), 0.2*(i % 4), 0.4*(i % 3)) for i in range(6)]
+
+for i, cont in enumerate(contributions):
+    if(i == 0):
+        label = "Trap 1"
+    else:
+        label = f"Trap D{i}"
+
+    plt.plot(temp[1:], cont, linestyle="--", color=colors[i], label=label)
+    plt.fill_between(temp[1:], 0, cont, facecolor="grey", alpha=0.1)
+
+
+# plotting original data
+experimental_tds = np.genfromtxt("dark-original.csv", delimiter=",")
+experimental_temp = experimental_tds[:, 0]
+experimental_flux = experimental_tds[:, 1]
+plt.scatter(experimental_temp, experimental_flux, color="black", label="original", s=16)
+
+plt.legend()
+plt.xlim(min_temp, max_temp)
+plt.ylim(bottom=0, top=1e17)
+plt.ylabel(r"Desorption flux (m$^{-2}$ s$^{-1}$)")
+plt.xlabel(r"Temperature (K)")
+
+plt.show()
+```
+
+```{note}
+The experimental data was taken from {cite}`dark_modelling_2024` using [WebPlotDigitizer](https://automeris.io/).
+```
```

Comment:
You can use the actual data from [this folder](https://github.com/J-Dark-PhD/Modelling_neutron_damage_effects_in_tungsten/tree/v1.0/section_3_model_parameter_evaluation/data/tds_data_schwartz_selinger)